x-pack/plugins/security_solution/common/detection_engine/schemas/common/rule_monitoring.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as t from 'io-ts';
+import { enumeration, IsoDateString, PositiveInteger } from '@kbn/securitysolution-io-ts-types';
+
+// -------------------------------------------------------------------------------------------------
+// Rule execution status
+
+/**
+ * Custom execution status of Security rules that is different from the status
+ * used in the Alerting Framework. We merge our custom status with the
+ * Framework's status to determine the resulting status of a rule.
+ */
+export enum RuleExecutionStatus {
+  'succeeded' = 'succeeded',
+  'failed' = 'failed',
+  'going to run' = 'going to run',
+  'partial failure' = 'partial failure',
+  /**
+   * @deprecated 'partial failure' status should be used instead
+   */
+  'warning' = 'warning',
+}
+
+export const ruleExecutionStatus = enumeration('RuleExecutionStatus', RuleExecutionStatus);
+
+export const ruleExecutionStatusOrder = PositiveInteger;
+export type RuleExecutionStatusOrder = t.TypeOf<typeof ruleExecutionStatusOrder>;
+
+// -------------------------------------------------------------------------------------------------
+// Rule execution metrics
+
+export const durationMetric = PositiveInteger;
+export type DurationMetric = t.TypeOf<typeof durationMetric>;
+
+export const ruleExecutionMetrics = t.partial({
+  total_search_duration_ms: durationMetric,
+  total_indexing_duration_ms: durationMetric,
+  execution_gap_duration_s: durationMetric,
+});
+
+export type RuleExecutionMetrics = t.TypeOf<typeof ruleExecutionMetrics>;
+
+// -------------------------------------------------------------------------------------------------
+// Rule execution summary
+
+export const ruleExecutionSummary = t.type({
+  last_execution: t.type({
+    date: IsoDateString,
+    status: ruleExecutionStatus,
+    status_order: ruleExecutionStatusOrder,
+    message: t.string,
+    metrics: ruleExecutionMetrics,
+  }),
+});
+
+export type RuleExecutionSummary = t.TypeOf<typeof ruleExecutionSummary>;
+
+// -------------------------------------------------------------------------------------------------
+// Rule execution events
+
+export const ruleExecutionEvent = t.type({
+  date: IsoDateString,
+  status: ruleExecutionStatus,
+  message: t.string,
+});
```

Comment:
Should it also contain `status_order: ruleExecutionStatusOrder`?

Reply:
I don't think so, at least it's not needed for showing last 5 failures. It might become clearer when we start working on https://github.com/elastic/kibana/issues/119598. We'll need to make changes in `getRuleExecutionEventsRoute` and maybe change this model as well.